script.js

```diff
@@ -1,4 +1,230 @@
+// Constants
+const CHOOSE_NUM_PLAYERS = "choose number of players"; // type of gameMode
+const CHOOSE_WIN_CNDTN = "choose the win condition"; // type of gameMode
+const CHOOSE_KNOCKOUT = "choose knockout mode"; // type of gameMode
+const CHOOSE_NUM_DICE = "choose number of dice for round"; // type of gameMode
+const HIGHEST = "highest"; // type of winCndtn
+const LOWEST = "lowest"; // type of winCndtn
+const ON = "on"; // type of koSetting
+const OFF = "off"; // type of koSetting
+
+// Global variables
+var gameMode = CHOOSE_NUM_PLAYERS; // Initialise game in this mode
+var numPlayers;
+var numDice;
+var winCndtn; // Highest or lowest wins?
+var koSetting; // Knockout Mode on or off?
+var players = []; // Array of player objects that will be updated as the game progresses
+
+// Generate a number between 1-6
+var rollDice = function () {
+  var randomInteger = Math.random() * 6;
+  var randomRoll = Math.floor(randomInteger) + 1;
+  return randomRoll;
+};
+
+// First stage of game: Choose number of players
+var stage1 = function (chosenNumber) {
+  var resultMessage;
+  // Input validation: need minimum 2 players to play. Limit number of players to 10 for performance reasons.
+  if (!(Number(chosenNumber) >= 2 && Number(chosenNumber) <= 10)) {
+    resultMessage = `You have entered an invalid input.<br>Please enter a number between 2 and 10. All decimals will be rounded up.`;
+    return resultMessage;
+  }
+  // Valid input, so we update numPlayers based on input rounded up
+  numPlayers = Math.ceil(Number(chosenNumber));
+  // Create object for each player and add to players array
+  for (var i = 0; i < numPlayers; i++) {
+    var player = { playerNum: i + 1, playerScore: 0 };
+    players.push(player);
+  }
+  // Output the result message
+  resultMessage = `You have selected to play the game with ${numPlayers} players.<br><br>Now enter "${HIGHEST}" to have the the win condition be ${"highest number wins".italics()}, or enter "${LOWEST}" to have the win condition be ${"lowest number wins".italics()}.`;
```

Comment:
I see that you have several long result messages. One thing you could do is create a helper function to return you a customized message based on the game mode and player. e.g. generateMessage(gameMode, player)